lib/dependencies.go

```diff
@@ -52,11 +55,21 @@ func (a *ACBuild) AddDependency(imageName, imageId string, labels types.Labels,
 		}
 	}()
 
-	acid, err := types.NewACIdentifier(imageName)
+	app, err := discovery.NewAppFromString(imageName)
```

Comment:
Sorry for not catching this earlier, but I guess this is actually the wrong approach to use this here - it should be in acbuild/dependency.go instead. That way we can retain the actual signature of this function where imageName is really an imageName, and labels is a definitive set of labels. Make sense?